c7n/resources/ec2.py

```diff
@@ -330,7 +330,11 @@ def __call__(self, i):
                 for a in v.get('Attachments', []):
                     if a['Device'] in self.skip:
                         volumes.remove(v)
-        return self.operator(map(self.match, volumes))
+        if self.operator(map(self.match, volumes)):
+            i['Volumes'] = volumes
```

Comment:
this isn't putting matched volumes its simply putting all volumes.

Reply:
Exact, this is what I described:
... a list of _all_ volumes attachment
... I did not find an easy way to only attach matched volumes